### ISSUE TITLE
Test method runs twice when used together with rstest

### ISSUE THREAD

**Issue by Aetf** [google/googletest-rust#77]
To reproduce:

`Cargo.toml`:

```toml
[package]
name = "double"
version = "0.1.0"
edition = "2021"

[dev-dependencies]
googletest = "0.3.0"
rstest = "0.16.0"
```

`src/main.rs`:

```rust
#[cfg(test)]
mod tests {
    use rstest::*;
    use googletest::{google_test, Result};

    #[rstest]
    #[google_test]
    fn this_runs_twice() -> Result<()> {
        println!("Hello");
        Ok(())
    }

    #[google_test]
    #[rstest]
    fn this_runs_once_with_warning() -> Result<()> {
        println!("Hello");
        Ok(())
    }
}
```

Output:
```
$ cargo test -- --show-output
warning: cannot test inner items
  --> src/main.rs:14:5
   |
14 |     #[rstest]
   |     ^^^^^^^^^
   |
   = note: `#[warn(unnameable_test_items)]` on by default
   = note: this warning originates in the attribute macro `test` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: `double` (bin "double" test) generated 1 warning
    Finished test [unoptimized + debuginfo] target(s) in 0.02s
     Running unittests src/main.rs (target/debug/deps/double-740489cfb1a1ebe7)

running 3 tests
test tests::this_runs_once_with_warning ... ok
test tests::this_runs_twice ... ok
test tests::this_runs_twice ... ok

successes:

---- tests::this_runs_once_with_warning stdout ----
Hello

---- tests::this_runs_twice stdout ----
Hello

---- tests::this_runs_twice stdout ----
Hello


successes:
    tests::this_runs_once_with_warning
    tests::this_runs_twice
    tests::this_runs_twice

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

Are these the expected result? From https://github.com/google/googletest-rust#interoperability I'd expect the test method to run once without any warnings regardless of the attribute ordering.

**Comment by hovinen** [comment]
Hi, thanks for reporting this!

There appears to be some issue with how the `rstest` and `googletest` procedural macros interact. I'll investigate this shortly.

**Comment by hovinen** [comment]
I've found the underlying problem and begun to work on a fix in the branch `fix-macro-interop`. It's not quite done yet. I need to add proper integration testing, make sure all the common libraries we need are supported, and so on. But it's available if you'd like to try it out.

**Comment by Aetf** [comment]
Hey, thanks for the quick fix! The warning is indeed gone.

I do notice a few more issues, though.

* `matches_pattern`/`pat` isn't working. This is reproducible with the new 0.4 release. Filed #86 for this.

* `fix-macro-interop` breaks fixtures in rstest. Try compile this:
```
#[cfg(test)]
mod tests {
    use googletest::{expect_that, google_test, Result};
    use rstest::*;

    struct FileFixture {
    }

    #[fixture]
    fn a_fixture() -> FileFixture {
        todo!()
    }

    #[google_test]
    #[rstest]
    fn with_fixture(a_fixture: FileFixture) -> Result<()> {
        todo!()
    }
}
```

The error is
```
error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> src/main.rs:16:8
   |
14 |     #[google_test]
   |     -------------- an argument of type `FileFixture` is missing
15 |     #[rstest]
16 |     fn with_fixture(a_fixture: FileFixture) -> Result<()> {
   |        ^^^^^^^^^^^^
   |
note: function defined here
  --> src/main.rs:16:8
   |
16 |     fn with_fixture(a_fixture: FileFixture) -> Result<()> {
   |        ^^^^^^^^^^^^ ----------------------
help: provide the argument
   |
14 |     with_fixture(/* FileFixture */)
   |

For more information about this error, try `rustc --explain E0061`.
```

**Comment by hovinen** [comment]
Hi, quick update on this: I've been working on a solution on branch `test_517440786`. It should at least fix things when `#[rstest]` appears *after* `#[google_test]`, including tests with parameters.

I'm not sure whether it's possible to fix things when the order is reversed. I'll give that some more thought but I might have to just make a note in the documentation.

Could you give that branch a try and let me know whether it works for you? If so, I'll try to push a small release with the fix next week. Thanks!

**Comment by Aetf** [comment]
Thanks for working on this! I can confirm that `test_517440786` works well when `#[rstest]` comes after `#[google_test]`, and fixtures are working.

**Comment by hovinen** [comment]
I've just released 0.4.2 which includes the (partial) fix for this issue. Thanks for reporting it!
